jupyter-extension/jupyterlab_pachyderm/pps_client.py

```diff
@@ -17,14 +17,49 @@
 from .log import get_logger
 
 METADATA_KEY = 'pachyderm_pps'
-
+PIPELINE_TYPE_SERVICE = 3
 
 class PPSClient:
     """Client interface for the PPS extension backend."""
 
     def __init__(self):
         self.nbconvert = PythonExporter()
 
+    async def retrieve(self, name, project_name):
+        """Retrieves details about the specified pipeline.
+
+        Args:
+            name: The name of the pipeline
+            project_name: The name of the project
+        """
+        get_logger().debug(f"Retrieving {project_name}/{name}")
+        client = self._client()
+
+        try:
+            for p in client.inspect_pipeline(pipeline_name=name, project_name=project_name, details=True):
+                # This is an incomplete serialization, as
+                # we only need the LB IP at present.
+                pd = {
+                    "pipeline": {
+                        "name": p.pipeline.name,
+                        "project": {
+                            "name": p.pipeline.project.name,
+                        },
+                    },
+                    "details": {}
+                }
+                if p.type == PIPELINE_TYPE_SERVICE:
+                    pd["details"]["service"] = {
+                        "internal_port": p.details.service.internal_port,
+                        "external_port": p.details.service.external_port,
+                        "ip": p.details.service.ip,
+                        "type": p.details.service.type,
+                    }
+                return json.dumps(pd)
+        except Exception as e:
+            get_logger().error(f"e: {e} pipeline: {p}")
+            raise e
```

Comment:
I think this should be raised as a `HTTPError`. The other methods should have this code for you to reference.